### PR TITLE
Make database calls generic

### PR DESCRIPTION
**Pull request by carlosljr**
The changes in this P.R. make HuskyCI support any kind of database since all calls for data retrieve will be doing with an interface.

With this improvement, any implementation of `Requests` interface will be accepted. However, it needs to be defined on the `context` package so it can return the new struct.

These changes are to attend the support for Postgres DB. We made some initial functions for the Postgres and insert its plugin.

This PR will also change the following environment variables: 
* `HUSKYCI_DATABASE_MONGO_ADDR` -> `HUSKYCI_DATABASE_DB_ADDR`
* `HUSKYCI_DATABASE_MONGO_DBNAME` -> `HUSKYCI_DATABASE_DB_NAME`
* `HUSKYCI_DATABASE_MONGO_DBUSERNAME` -> `HUSKYCI_DATABASE_DB_USERNAME`
* `HUSKYCI_DATABASE_MONGO_DBPASSWORD` -> `HUSKYCI_DATABASE_DB_PASSWORD`

### CODE REVIEW

File: api/db/mongo/mongo.go [hunk 1]
```diff
@@ -41,18 +41,19 @@ type Database interface {
 }
 
 // Connect connects to mongo and returns the session.
-func Connect() error {
+func Connect(address, dbName, username, password string, poolLimit, port int, timeout time.Duration) error {
 
-	log.Info("Connect", "DB", 21)
-	mongoConfig := config.APIConfiguration.MongoDBConfig
+	values := fmt.Sprintf("Mongo received vars: Address - %s\nDBName - %s\nUsername - %s\nPassword - %s\nPoolLimit - %d\nPort - %d \nTimeout - %v\n", address, dbName, username, password, poolLimit, port, timeout)
```
Comment (rafaveira3) [inline api/db/mongo/mongo.go]: Ops, it looks like sensitive data is being sent to the logging system! 😅 Would you mind removing it, please?

File: api/db/postgres/postgres_handler.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+	"time"
+	_ "github.com/lib/pq" // Defining postgres plugin
```
Comment (rafaveira3) [inline api/db/postgres/postgres_handler.go]: Would you mind removing this comment, please? 🙃

File: api/db/postgres/postgres_handler.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+// ConfigureDB will establish a new connection
+// with the postgres DB referenced in the arguments
+func (pHandler *PostgresHandler) ConfigureDB(
+	address string,
+	username string,
+	password string,
+	dbName string) error {
+	doOnce.Do(func() {
+		connStr := fmt.Sprintf("postgres://%s:%s@%s/%s?sslmode=disable",
```
Comment (rafaveira3) [inline api/db/postgres/postgres_handler.go]: What do you think of adding `sslmode` as a configurable option as well? By the way, are we expecting this `disable` to be the default value? 🧐

File: api/db/postgres/postgres_handler.go (hunk 2 above)
Comment (carlosljr) [reply api/db/postgres/postgres_handler.go]: It is necessary for the lint. You need to justify the if you import a package with `_` reference :(

The following message is generated if I remove the comment:
```
exported method PostgresRequests.ConnectDB should have comment or be unexported
```
I don't have option not to import the psql plugin. However, I don't it explictly. So I need to solve this using the comment.

File: api/context/context.go [hunk 4]
```diff
@@ -188,53 +194,92 @@ func (dF DefaultConfig) GetGraylogIsDev() bool {
...
-		Username:     dF.Caller.GetEnvironmentVariable("HUSKYCI_DATABASE_MONGO_DBUSERNAME"),
-		Password:     dF.Caller.GetEnvironmentVariable("HUSKYCI_DATABASE_MONGO_DBPASSWORD"),
-		Port:         mongoPort,
-		Timeout:      dF.GetMongoTimeout(),
-		PoolLimit:    dF.GetMongoPoolLimit(),
+func (dF DefaultConfig) getDBConfig() *DBConfig {
+	dbHost := dF.Caller.GetEnvironmentVariable("HUSKYCI_DATABASE_DB_ADDR")
+	dbPort := dF.GetDBPort()
+	return &DBConfig{
+		Address:         dbHost,
```
[the remark this answers is not in the document]
Comment (carlosljr) [reply api/context/context.go]: Ok! Changes have been made in this (commit)[https://github.com/globocom/huskyCI/pull/400/commits/a4a6afe14bd719d3281c3d1de12be4a32b31e575]

File: api/context/context.go [hunk 5]
```diff
@@ -78,6 +82,7 @@ type APIConfig struct {
 	YarnAuditSecurityTest  *types.SecurityTest
 	GitleaksSecurityTest   *types.SecurityTest
 	SafetySecurityTest     *types.SecurityTest
+	DbInstance             db.Requests
```
[the remark this answers is not in the document]
Comment (carlosljr) [reply api/context/context.go]: Ok! Changes have been made in this [commit](https://github.com/globocom/huskyCI/pull/400/commits/a4a6afe14bd719d3281c3d1de12be4a32b31e575).

File: api/context/context.go [hunk 6]
```diff
@@ -304,3 +349,16 @@ func (dF DefaultConfig) GetMaxContainersAllowed() int {
...
+
+// GetDB returns a Requests implementation based on the
+// on the type configured on HUSKYCI_DATABASE_TYPE env var.
+// The default returns a MongoRequests that implements mongo
+// queries.
+func (dF DefaultConfig) GetDB() db.Requests {
+	dB := dF.Caller.GetEnvironmentVariable("HUSKYCI_DATABASE_TYPE")
+	if strings.EqualFold(dB, "postgres") {
+		return nil
+		// return &db.PostgresRequests{}
```
[the remark this answers is not in the document]
Comment (carlosljr) [reply api/context/context.go]: Ok! Changes have been made in this [commit](https://github.com/globocom/huskyCI/pull/400/commits/a4a6afe14bd719d3281c3d1de12be4a32b31e575).

File: api/db/mongo/mongo.go (hunk 1 above)
Comment (carlosljr) [reply api/db/mongo/mongo.go]: Ok! Changes have been made in this [commit](https://github.com/globocom/huskyCI/pull/400/commits/a4a6afe14bd719d3281c3d1de12be4a32b31e575).

File: api/db/postgres/postgres_handler.go (hunk 3 above)
Comment (rafaveira3) [reply api/db/postgres/postgres_handler.go]: Absolutely!

**Pull request merged**